### PR TITLE
[azopenai] Adding in streaming for ChatCompletions, like we have for Completions.

### LINKED ISSUE #21052: [azopenai] API modification feedback
Create GetChatCompletionsStream, similar to what we did for GetCompletionsStream().

### PR DESCRIPTION
**Pull request by richardpark-msft**
Also, did some cleanup/moving around of code to make it easier to find tests, etc...

Fixes #21052

### CODE REVIEW

**Comment by richardpark-msft** [comment]
CC: @JeffreyRichter

File: sdk/cognitiveservices/azopenai/custom_client.go [hunk 1]
```diff
@@ -109,17 +109,20 @@ func (b *openAIPolicy) Do(req *policy.Request) (*http.Response, error) {
...
-
 type streamCompletionsOptions struct {
-	CompletionsOptions
+	// we strip out the 'stream' field from the options exposed to the customer so
+	// now we need to add it back in.
+	any
 	Stream bool `json:"stream"`
 }
 
 func (o streamCompletionsOptions) MarshalJSON() ([]byte, error) {
```
Comment (RickWinter) [inline sdk/cognitiveservices/azopenai/custom_client.go:119]: Should MarshalJSON be exposed to customers?

**Review by RickWinter — APPROVED** [review]

**Review by mikekistler — APPROVED** [review]
Looks good! 👍

File: sdk/cognitiveservices/azopenai/custom_client.go (hunk 1 above)
Comment (JeffreyRichter) [reply sdk/cognitiveservices/azopenai/custom_client.go:119]: No, but Go gives us no choice in the matter - it must be exposed.

File: sdk/cognitiveservices/azopenai/client_chat_completions_test.go [hunk 2]
```diff
@@ -0,0 +1,187 @@
+//go:build go1.18
```
[the remark this answers is not in the document]
Comment (richardpark-msft) [reply sdk/cognitiveservices/azopenai/client_chat_completions_test.go:1]: I have a few basic ones, but that's a great idea. I'll add some more in another PR.

**Pull request merged**